torchbenchmark/models/dlrm/__init__.py

```diff
@@ -163,8 +163,8 @@ def __init__(self, device=None, jit=False):
 
         # Preparing data
         X, lS_o, lS_i, self.targets = next(iter(self.train_ld))
+        X = X.to(self.device)
         if self.device == "cuda":
-            X = X.to(self.device)
             lS_i = [S_i.to(self.device) for S_i in lS_i] if isinstance(lS_i, list) \
```

Comment:
don't we also need to move S_i and S_o to lazy device?

maybe its better to just change the conditional to 
`if self.device != "cpu":`
?

Reply:
Fixed.